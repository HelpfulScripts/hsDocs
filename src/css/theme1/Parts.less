.hsdocs_padding {
    padding: 10px 0.4em 2px 0em;  
}

@TitleBackground: #eef;
@TitleBorder: darken(@TitleBackground, 5%);

.hsdocs_title {    // applies to the title row
    // background-color: @TitleBackground;
    // border-top: 1px solid @TitleBorder;
    // border-bottom: 1px solid @TitleBorder;
    // margin-top: 5px;
    .hsdocs_padding;
    padding-left: 5px;
}

.hsdocs_itemname {
    font-weight: bold;
    color: #008;
    font-size: 1.1em;
}

.hsdocs_type_unknown {
    color: #f44;
}



.hsdocs_flag {
    color: #44f;
    .hsdocs_padding
}

.hsdocs_flag_export {
    .hsdocs_flag;
}

.hsdocs_flag_public {
    .hsdocs_flag;
    color: #242;
}

.hsdocs_flag_private {
    .hsdocs_flag;
    color: #844;
}

.hsdocs_flag_protected {
    .hsdocs_flag;
    color: #884;
}

.hsdocs_flag_static {
    .hsdocs_flag;
    color: #666;
}

.hsdocs_flag_abstract {
    .hsdocs_flag;
    color: #888;
}

.hsdocs_flag_const {
    .hsdocs_flag;
    font-weight: bold;
    color: #222;
}

.hsdocs_flag_let {
    .hsdocs_flag;
    font-weight: bold;
    color: #222;
}

.hsdocs_flag_var {
    .hsdocs_flag;
    font-weight: bold;
    color: #222;
}

.hsdocs_flag_constructor-property {
    .hsdocs_flag;
    color: #848;
}

.hsdocs_flag_optional {
    .hsdocs_flag;
    color: #aaa;
}

.hsdocs_flag_unknown {
    .hsdocs_flag;
    color: #f44;
    font-weight: bold;
}

.hsdocs_flag_ignore {
    .hsdocs_flag;
}

.hsdocs_flag_optional {
    &:after { 
        content: '?'; 
        font-style: italic;
        font-weight:bold;
    }
}



.hsdocs_kind {
    font-style: italic;
    .hsdocs_padding;
}


.hsdocs_extends {
    .hsdocs_padding;
    padding-left: 0.3em;
}

.hsdocs_extensionOf {
    font-size: 80%;
    .hsdocs_padding;
}

.hsdocs_extensionType {
    .hsdocs_padding;
}

.hsdocs_implementationOf {
    font-size: 80%;
    .hsdocs_padding;
    font-style: italic;
}

.hsdocs_implements { 
    .hsdocs_padding;
}

.hsdocs_extended_by {
    .hsdocs_padding;
    // font-size: 80%;
}

.hsdocs_implemented_by {
    .hsdocs_padding;
    // font-size: 80%;
}

.hsdocs_type_unknown {
    color: #f44;
}

.hsdocs_signature {
    font-size: 1em;
    .hsdocs_itemname {
        padding:0;
    }
}

@ItemBorderRadius: 4px;

.hsdocs_type_default {
    font-size: 90%;
    background-color: #88f;
    color: #fff;
    padding: 2px;
    border-radius: @ItemBorderRadius;
//    font-style: italic;
    font-weight: bold;
}

.hsdocs_type {
    // border-radius: @ItemBorderRadius;
    // background-color: #ddb;
    // padding:  2px 4px;
    // margin-left: 2px;
    // margin-right: 2px;
    // font-size: 0.9em;
    // :link {
    //     text-decoration: none;
    // }
}

.hsdocs_type_instrinct {
    font-style: italic;
    color: #800;
}

.hsdocs_signature_param {
    margin: 0 2px;
    .hsdocs_itemname {
        color: #008;
        font-size: 1em;
        font-weight:normal;
    }
}

.hsdocs_source {
    float:right;
    padding: 5px 0em;  
    margin-right: 10px;
    bottom: 3px;
    font-size: 0.8rem;
}

