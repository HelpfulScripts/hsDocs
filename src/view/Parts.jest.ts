import { flags } from './Parts';
import { kindString } from './Parts';
import { itemName } from './Parts';
import { itemTooltip } from './Parts';
import { extensionOf } from './Parts';
import { inheritedFrom } from './Parts';
import { sourceLink } from './Parts';
import { libLink } from './Parts';
import { signature } from './Parts';
import { defaultVal } from './Parts';
import { type } from './Parts';
import { makeID } from './Parts';

test('flags', ()=>{
    expect(flags).toBeDefined();
});
test('kindString', ()=>{
    expect(kindString).toBeDefined();
});
test('itemName', ()=>{
    expect(itemName).toBeDefined();
});
test('itemTooltip', ()=>{
    expect(itemTooltip).toBeDefined();
});
test('extensionOf', ()=>{
    expect(extensionOf).toBeDefined();
});
test('inheritedFrom', ()=>{
    expect(inheritedFrom).toBeDefined();
});
test('sourceLink', ()=>{
    expect(sourceLink).toBeDefined();
});
test('libLink', ()=>{
    expect(libLink).toBeDefined();
});
test('signature', ()=>{
    expect(signature).toBeDefined();
});
test('defaultVal', ()=>{
    expect(defaultVal).toBeDefined();
});
test('type', ()=>{
    expect(type).toBeDefined();
});
test('makeID', ()=>{
    expect(makeID).toBeDefined();
});
