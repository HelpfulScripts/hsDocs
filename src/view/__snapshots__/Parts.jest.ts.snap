// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parts extensionOf creates 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-extensions",
  },
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-extends",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "extends",
    },
    Object {
      "attrs": null,
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-extension",
          },
          "children": Array [
            Object {
              "attrs": Object {
                "href": "#!/api/hsDocs/hsDocs.Site.SiteName",
                "oncreate": undefined,
                "onupdate": undefined,
              },
              "children": undefined,
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "a",
              "text": "id275",
            },
            Object {
              "attrs": undefined,
              "children": "",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts flags creates 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-flags",
  },
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-export-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "export",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-external-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "external",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-public-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "public",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-constructorProperty-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "constructorProperty",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-abstract-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "abstract",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-const-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "const",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-static-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "static",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-optional-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "optional",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-unknown-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "isUndefined",
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts flags creates ignoring const 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-flags",
  },
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-export-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "export",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-external-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "external",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-public-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "public",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-constructorProperty-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "constructorProperty",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-abstract-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "abstract",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-ignore-flag",
      },
      "children": Array [],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    Object {
      "attrs": Object {
        "className": "hs-item-static-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "static",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-optional-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "optional",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-unknown-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "isUndefined",
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts flags creates ignoring export 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-flags",
  },
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-ignore-flag",
      },
      "children": Array [],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    Object {
      "attrs": Object {
        "className": "hs-item-external-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "external",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-public-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "public",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-constructorProperty-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "constructorProperty",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-abstract-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "abstract",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-const-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "const",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-static-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "static",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-optional-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "optional",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-unknown-flag",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "isUndefined",
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts flags creates missing flags 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-flags",
  },
  "children": Array [],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts inheritedFrom creates 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-inherited-from",
  },
  "children": Array [
    Object {
      "attrs": null,
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": "inherited from ",
    },
    Object {
      "attrs": Object {
        "href": "#!/api/hsDocs/hsDocs.Site",
        "oncreate": undefined,
        "onupdate": undefined,
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "a",
      "text": "Site",
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts itemTooltip creates 1`] = `
Object {
  "attrs": Object {
    "className": "hs-item-name",
  },
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-tooltip",
        "href": "#",
      },
      "children": Array [
        Object {
          "attrs": undefined,
          "children": "test",
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "#",
          "text": undefined,
        },
        Object {
          "attrs": Object {
            "className": "hs-tooltip-bottom",
          },
          "children": undefined,
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": "class name and then some",
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "div",
      "text": undefined,
    },
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates array 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-array",
          },
          "children": Array [
            Object {
              "attrs": undefined,
              "children": "Array<",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
            Object {
              "attrs": undefined,
              "children": "",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
            Object {
              "attrs": undefined,
              "children": ">",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates default 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "error",
          },
          "children": undefined,
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": "bogus",
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates instrinct 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-instrinct",
          },
          "children": Array [
            Object {
              "attrs": Object {
                "href": "#!/api/hsDocs/undefined",
                "oncreate": undefined,
                "onupdate": undefined,
              },
              "children": Array [
                null,
              ],
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "span",
              "text": undefined,
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates reference 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-reference",
          },
          "children": Array [
            Object {
              "attrs": Object {
                "href": "#!/api/hsDocs/hsDocs.view.MainExample.modules.hsdatab",
                "oncreate": undefined,
                "onupdate": undefined,
              },
              "children": undefined,
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "a",
              "text": "hsdatab",
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates reflection 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-reflection",
          },
          "children": Array [
            Object {
              "attrs": Object {
                "className": "hs-item-reflection",
              },
              "children": Array [
                Object {
                  "attrs": undefined,
                  "children": "{ ",
                  "dom": undefined,
                  "domSize": undefined,
                  "events": undefined,
                  "instance": undefined,
                  "key": undefined,
                  "state": undefined,
                  "tag": "#",
                  "text": undefined,
                },
                Object {
                  "attrs": undefined,
                  "children": " }",
                  "dom": undefined,
                  "domSize": undefined,
                  "events": undefined,
                  "instance": undefined,
                  "key": undefined,
                  "state": undefined,
                  "tag": "#",
                  "text": undefined,
                },
              ],
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "span",
              "text": undefined,
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates stringLiteral 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-string-literal",
          },
          "children": undefined,
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": "stringLiteral",
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates tuple 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-tuple",
          },
          "children": Array [
            Object {
              "attrs": undefined,
              "children": "[ ",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
            Object {
              "attrs": undefined,
              "children": Array [
                null,
                Object {
                  "attrs": undefined,
                  "children": "",
                  "dom": undefined,
                  "domSize": undefined,
                  "events": undefined,
                  "instance": undefined,
                  "key": undefined,
                  "state": undefined,
                  "tag": "#",
                  "text": undefined,
                },
              ],
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "[",
              "text": undefined,
            },
            Object {
              "attrs": undefined,
              "children": " ]",
              "dom": undefined,
              "domSize": undefined,
              "events": undefined,
              "instance": undefined,
              "key": undefined,
              "state": undefined,
              "tag": "#",
              "text": undefined,
            },
          ],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;

exports[`parts type creates union 1`] = `
Object {
  "attrs": null,
  "children": Array [
    Object {
      "attrs": Object {
        "className": "hs-item-name",
      },
      "children": undefined,
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": ":",
    },
    Object {
      "attrs": Object {
        "className": "hs-item-sig-type",
      },
      "children": Array [
        Object {
          "attrs": Object {
            "className": "hs-item-type-union",
          },
          "children": Array [],
          "dom": undefined,
          "domSize": undefined,
          "events": undefined,
          "instance": undefined,
          "key": undefined,
          "state": undefined,
          "tag": "span",
          "text": undefined,
        },
      ],
      "dom": undefined,
      "domSize": undefined,
      "events": undefined,
      "instance": undefined,
      "key": undefined,
      "state": undefined,
      "tag": "span",
      "text": undefined,
    },
    null,
  ],
  "dom": undefined,
  "domSize": undefined,
  "events": undefined,
  "instance": undefined,
  "key": undefined,
  "state": undefined,
  "tag": "span",
  "text": undefined,
}
`;
